libraries/AP_BattMonitor/AP_BattMonitor_FuelLevel_Analog.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Code by Charlie Johnson
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_BattMonitor_FuelLevel_Analog.h"
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_BattMonitor_FuelLevel_Analog::var_info[] = {
+
+    // @Param: FL_VLT_MIN
+    // @DisplayName: Empty fuel level voltage
+    // @Description: The voltage seen on the analog pin when the fuel tank is empty. Note: For this type of battery monitor, the voltage seen by the analog pin is displayed as battery voltage on a GCS.
+    // @Range: 0.01 10
+    // @Units: V
+    // @User: Advanced
+    AP_GROUPINFO("FL_VLT_MIN", 50, AP_BattMonitor_FuelLevel_Analog, _fuel_level_empty_voltage, 0.5),
```

Comment:
Is there a backend using 40-49? Already? I didn't see it. 

We should probably add a comment table for this somewhere higher up?
We are near the end soon and will need to extend it. 

```suggestion
    AP_GROUPINFO("FL_VLT_MIN", 40, AP_BattMonitor_FuelLevel_Analog, _fuel_level_empty_voltage, 0.5),
```

Reply:
i looked and didn't find a 40-49 in the backends of master? 

By comment table somehwere that says

```cpp
// Paramter Indices for Parameters Loaded By Pointer
// Analog 1-9
// SMBUS 10-19
// INA2xx.cpp 25-26 (comments for this and below need to be fixed)
// UAVCAN 20-29
// Sum 30-39
// Fuel_Level_Analog 40-49
```

Reply:
Fixed. Figured out where I got confused. The 40 range was used for the AP_SUBGROUPVARPTR instances.